hazelcast/src/main/java/com/hazelcast/cache/impl/ICacheService.java

```diff
@@ -90,4 +91,8 @@
     String addInvalidationListener(String name, CacheEventListener listener);
 
     void sendInvalidationEvent(String name, Data key, String sourceUuid);
+
+    boolean wanReplicationEnabled(String cacheName);
```

Comment:
Yes, `ICacheRecordStore` is specific to a cache so there is no need to lookup every time if we check the wan replication enabled status over  `ICacheRecordStore`. I mean you can use a final boolean variable such as `wanReplicationEnable` and initialize it at constructor of `ICacheRecordStore` by calling `ICacheService::wanReplicationEnabled`. So there will be no map lookup every time in this case and this will be faster. 

However, please keep the `wanReplicationEnabled` in the `ICacheService` also I prefer `isWanReplicationEnable` as method name instead of `wanReplicationEnabled`.